dask/async.py

```diff
@@ -473,7 +473,12 @@ def fire_task():
 
     # Main loop, wait on tasks to finish, insert new ones
     while state['waiting'] or state['ready'] or state['running']:
-        key, res, tb, worker_id = queue.get()
+        try:
+            key, res, tb, worker_id = queue.get()
+        except KeyboardInterrupt:
+            for f in finish_cbs:
+                f(dsk, state, True)
+            raise
```

Comment:
Yeah, this ended up working cleanly in a few different contexts.  It also feels cleaner than emitting the usual monster traceback which, in this context, I don't think we want.